src/librustc_codegen_llvm/llvm/mod.rs

```diff
@@ -87,6 +87,7 @@ pub struct RustString {
 }
 
 /// Appending to a Rust string -- used by RawRustStringOstream.
+#[allow(improper_ctypes)]
 #[no_mangle]
 pub unsafe extern "C" fn LLVMRustStringWriteImpl(sr: &RustString,
```

Comment:
I wonder if there is a pattern that we'd officially support for stuff like this: `RustString` is opaque to C++ and always behind a reference (passed from Rust to begin with), so the field types or size don't matter for FFI.

Come to think of it, we shouldn't be declaring `no_mangle` symbols at all, but rather passing a `fn` pointer alongside the buffer, for writing to it. And maybe use `restrict` instead of a `RefCell`.

Reply:
Interesting ideas, I want to try some of them, but let's open an issue for that since it's unrelated to this PR and out-of-scope.